File: src/main/julia/day05.jl
include("intcomp.jl")

prog = map(x->parse(Int64, x), split(readlines("input05.txt")[1], ","))

# part 1

c = IntComputer(prog)
c.output = println
c.input = ()->1
@time run!(c)

# part 2

# c = IntComputer([3,21,1008,21,8,20,1005,20,22,107,8,21,20,1006,20,31, 1106,0,36,98,0,0,1002,21,125,20,4,20,1105,1,46,104, 999,1105,1,46,1101,1000,1,20,4,20,1105,1,46,98,99])
# c.output = println
# c.input = ()->9
# run(c)

c = IntComputer(prog)
c.output = println
c.input = ()->5
@time run!(c)

File: src/main/python/day05.py

from intcomp import IntComputer

with open("input05.txt") as f:
    prog = [int(x) for x in f.readlines()[0].split(",")] 

c = IntComputer([3,15,3,16,1002,16,10,16,1,16,15,15,4,15,99,0,0])
c.input = lambda : 1
c.run()

# part 2

c = IntComputer(prog)
c.input = lambda : 5
c.run()
